src/Events/QR/QR_Code.php

```diff
@@ -52,6 +52,17 @@ class QR_Code {
 	 */
 	private $qr_url;
 
+	/**
+	 * Check if QR functionality is enabled.
+	 *
+	 * @since TBD
+	 * @return bool Whether QR functionality is enabled.
+	 */
+	public function is_enabled(): bool {
```

Comment:
Personally, I'd put the check in the controller is_active() and base everything on that (except the things you _need_ like setting to turn it back on)

i.e. "Shut them all down!" (sorry, random Star Wars quote)